c++/src/kj/common.h

```diff
@@ -206,6 +206,32 @@ typedef unsigned char byte;
 #define KJ_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
 #endif
 
+#ifdef __has_cpp_attribute
+#define KJ_HAS_CPP_ATTRIBUTE(x) __has_cpp_attribute(x)
+#else
+#define KJ_HAS_CPP_ATTRIBUTE(x) 0
+#endif
+
+#if KJ_HAS_CPP_ATTRIBUTE(clang::lifetimebound)
+#define KJ_LIFETIMEBOUND [[clang::lifetimebound]]
```

Comment:
I wonder if we'll need to give people a way to opt out of this? Since it did produce several false positives in our own code. Is it a warning by default, or an error?

Reply:
It's a warning. In our system I explicitly opted into having them be errors. I can change this to be an opt-in thing.

Reply:
Done. `KJ_ENABLE_LIFETIMEBOUND`

Reply:
It looks like you implemented an opt-in rather than an opt-out. To clarify I think this should be on by default, but I was concerned someone would whine to us that it breaks their code. OTOH they can presumably disable it using the appropriate `-w` flag so in retrospect I guess it probably doesn't need a flag at all. Sorry for not being cleaer.